python/foglamp/services/south/server.py

```diff
@@ -117,29 +122,34 @@ class Server(FoglampMicroservice):
 
             self._plugin_handle = self._plugin.plugin_init(config)
 
+            await Ingest.start(self._core_management_host, self._core_management_port)
+
             # Executes the requested plugin type
             if self._plugin_info['mode'] == 'async':
-                await  self._exec_plugin_async(config)
+                self._task_async = asyncio.ensure_future(self._exec_plugin_async())
             elif self._plugin_info['mode'] == 'poll':
-                asyncio.ensure_future(self._exec_plugin_poll(config))
-
-        except (Exception, KeyError) as ex:
+                self._task_poll = asyncio.ensure_future(self._exec_plugin_poll())
+        except asyncio.CancelledError:
+            pass
+        except exceptions.DataRetrievalError:
+            _LOGGER.exception('Data retreival error in plugin {}'.format(self._name))
+        except (Exception, KeyError, pexpect.exceptions.TIMEOUT) as ex:
```

Comment:
better to capture pexpect.exceptions.TIMEOUT at plugin level in timeouterror and raise that with info